readthedocs_ext/readthedocs.py

```diff
@@ -172,12 +182,61 @@ def generate_json_artifacts(app, pagename, templatename, context, doctree):
         log.exception(
             'Fail to save JSON output for page {page}'.format(page=outjson)
         )
-    except Exception as e:
+    except Exception:
         log.exception(
             'Failure in JSON search dump for page {page}'.format(page=outjson)
         )
 
 
+def dump_sphinx_data(app, exception):
+    """
+    Dump a bunch of additional Sphinx data that is useful during search indexing
```

Comment:
Docstring could be expanded with the output structure expected of the dict mentioning what is each field and some examples about how are the values (paths --to see if they are relative/absolute, etc) for them.